scale-tests/streaming_test.py

```diff
@@ -0,0 +1,102 @@
+import json
+import sys
+
+import sdk_cmd
+import spark_utils as utils
+
+KAFKA_PACKAGE_NAME = "kafka"
+KAFKA_SERVICE_NAME = "kafka"
+PRODUCER_SERVICE_NAME = "Spark->Kafka Producer"
+KERBEROS_FLAG = "false"
+JAR_URI = "https://s3-us-west-2.amazonaws.com/infinity-artifacts/soak/spark/dcos-spark-scala-tests-assembly-0.1-SNAPSHOT.jar"
+PRODUCER_AVG_NUM_WORDS_PER_MIN = 500
+
+
+def run_pipeline(num_consumers, desired_runtime, spark_app_name):
+    broker_dns = _kafka_broker_dns()
+    topic = "topic-{}".format(spark_app_name)
+
+    num_words_to_read = int(desired_runtime * PRODUCER_AVG_NUM_WORDS_PER_MIN / num_consumers)
```

Comment:
I don't understand this. Don't all the consumers consume from the beginning offset of the topic? (They all start at offset 0 and read to the end of the topic.) So, the number of words shouldn't depend on the number of consumers.

Reply:
just to keep record since we discussed it already - it doesn't seem like consumers all read from the topic at the same time. Producers write events and only 1 consumer reads at a time. When it's done, the next consumer in line starts reading. That's why this function is dependent on `num_consumers`. 

I'll look into why this is and see if we can get all the consumers to read at the same time